### ISSUE TITLE
Upload not working consistently on chrome on iOS

### ISSUE THREAD

**Issue by canvas-manik** [danialfarid/ng-file-upload#1217]
Hi Danial,
Thanks for providing the library. It really helps.

It is working fine with chrome on desktop, while chrome on iPad 4 and iPhone 5, sometimes it works, sometimes it gets stuck. I am sending the file to node api, where sometimes it is reaches, while sometimes it does not.

Looks like some memory leak issue, as if I try reloading the browser, it works while subsequent times, it gets stuck.
Are there some best practices for using ng-file-upload with mobile devices?
Thanks and regards,
Manik Mittal

**Comment by danialfarid** [comment]
If the file is too big it might run out of memory. Also make sure you don't keep a reference to the file model. If you have preview or thumbnail of the file you can remove that to make sure that doesn't cause this problem. 
If you could provide a jsfiddle with steps to reproduce would help to find the issue.

**Comment by canvas-manik** [comment]
Yes,the file is little big ~1.5 MB. which actually is the standard size what we get while we click from iPad / iPhone. I tried using your jsfiddle on http://jsfiddle.net/danialfarid/0mz6ff9o/135/ (I modified the jsfiddle to remove ngf-max-height="1000" ngf-max-size="1MB"). There also with desktop the progress bar is shown up, while on iPad it didn't. I tried with this photo,which is clicked from iPad.
![img_0434](https://cloud.githubusercontent.com/assets/11735090/11921492/e8b4a164-a7af-11e5-9072-10707ac3cf5b.JPG)

**Comment by danialfarid** [comment]
Is it just the progress?
Does the file actually get uploaded?

**Comment by canvas-manik** [comment]
Hi Danial,

Thanks again for the awesome package. I have created a jsfiddle on http://jsfiddle.net/2vq88rfs/294/
where I am trying to upload the files my api server. I see from my iPad, i tried uploading 6 images, it uploaded few images and then got stuck. Possibly, a mix of some memory leak and low memory availability on iOS Chrome.

I saw in other issue posted here: https://github.com/danialfarid/ng-file-upload/issues/1222. Tried changing to input type=file for directive on http://jsfiddle.net/2vq88rfs/294/. Then also,it did fail. 

Hope this information might help in debugging the issue.

Regards,
Manik Mittal
